feat: Add docs for custom browser tracing routing instrumentation (#9113)
getsentry/sentry-docs: platform-includes/performance/automatic-instrumentation-custom-routing/javascript.mdx

### Custom routing

By default, the `browserTracingIntegration()` will create a `pageload` span for when the page is initially loaded, as well as a `navigation` span whenever the URL changes afterwards.

If you want to ensure navigation spans are created correctly for a custom routing setup, you can opt out of the default `navigation` span creation by setting `instrumentNavigation: false` in the `browserTracingIntegration()` options. Then, you can manually create navigation spans like this:

```javascript
Sentry.init({
  integrations: [
    // disable default navigation spans
    Sentry.browserTracingIntegration({ instrumentNavigation: false }),
  ],
});

// Somewhere, instrument your router like this:

> myRouter.on("routeChange", (route) => {

Reviewer: We should add a note about setting correct route name for pageload as well, we want to avoid un-paramaterized transactions.

Author: I guess this is mostly about updating the pageload span, right? 🤔 or should we document how to emit a pageload span?

Reviewer: yeah updating the pageload span, I think most frameworks will be fine with emitting pageload as soon as possible.

Author: I opened a PR: https://github.com/getsentry/sentry-javascript/pull/10633 to make this easier, as documenting this showed that there is def. room for improvement there!

> ### Custom routing

Reviewer 2: ```suggestion
### Custom Routing
```

> By default, the `browserTracingIntegration()` will create a `pageload` span for when the page is initially loaded, as well as a `navigation` span whenever the URL changes afterwards.
> 
> If you want to ensure spans are created correctly for a custom routing setup, you can opt out of the default span creation by setting `instrumentNavigation: false` & `instrumentPageLoad: false` in the `browserTracingIntegration()` options. Then, you can manually create the spans like this:

Reviewer 2: ```suggestion
By default, the `browserTracingIntegration()` will create a `pageload` span for when the page is initially loaded, as well as a `navigation` span for whenever the URL changes afterwards.

To make sure that spans are created correctly for a custom routing setup, you'll need to opt out of the default span creation by setting `instrumentNavigation: false` and `instrumentPageLoad: false` in the `browserTracingIntegration()` options. You can then manually create spans like this:
```

```javascript
Sentry.init({
  integrations: [
    Sentry.browserTracingIntegration({
      // disable automatic span creation
      instrumentNavigation: false,
      instrumentPageLoad: false,
    }),
  ],
});

// We start the pageload span as early as possible!
let pageLoadSpan = Sentry.startBrowserTracingPageLoadSpan({
  name: window.location.pathname
  attributes: {
    [Sentry.SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: "url",
  },
});

// Somewhere, instrument your router like this:
myRouter.on("routeChange", (route) => {
  const client = Sentry.getClient();

>   // We want to make sure to update the pageload span to use the route name
>   // After that, each route change should trigger a navigation span (which will automatically finish the previous one)

Reviewer 2: ```suggestion
  // Make sure that the pageload span uses the route name
  // After that, each route change should trigger a navigation span (which will automatically finish the previous one)
```
